### DISCUSSION TITLE
How to setup my salt key and run mode on server?

### DISCUSSION THREAD

**Discussion by yahu1031** [serverpod/serverpod#686] [Q&A] [answered]
How to set my custom salt key and run mode on server part of code?

**Comment by fischerscode** [comment, marked as answer]
> How to set my custom salt key

Did not test this, but from having a look at the code, I assume you should be able to provide the salt by adding a `email_password_salt` entry to the `passwords.yaml` file. (Would be good to document this...)

<details>
<summary>Code:</summary>

https://github.com/serverpod/serverpod/blob/d86a69b781daa454788a440ff058606f77240edd/modules/serverpod_auth/serverpod_auth_server/lib/src/business/email_auth.dart#L12-L22
</details>

> and run mode on server part of code?

If you refer to the `production` mode, it can be set using [`--mode`](https://docs.serverpod.dev/deployments/general#configuration-files).
